tests/qs/utils.test.ts

```diff
@@ -167,3 +168,49 @@ test('is_buffer()', function () {
   // t.equal(is_buffer(buffer), true, 'real Buffer instance is a buffer');
   expect(is_buffer(buffer)).toEqual(true);
 });
+
+test("debug()", function(){
+  const originalEnv = process.env;
+  const spy = jest.spyOn(console, "log");
+
+  // Debug enabled
+  process.env["DEBUG"]= "true";
+
+  // Test request body includes headers object with Authorization
+  const headersTest = {
+    headers: {
+      Authorization: "fakeAuthorization"
+    }
+  }
+  debug("request", headersTest);
+  expect(spy).toHaveBeenCalledWith("OpenAI:DEBUG:request", {
+    headers: {
+      Authorization: "REDACTED"
+    }
+  });
+
+  // Test request body includes headers object with api-ley
+  const apiKeyTest = {
+    headers: {
+      "api-key": "fakeKey"
+    }
+  }
+  debug("request", apiKeyTest);
+  expect(spy).toHaveBeenCalledWith("OpenAI:DEBUG:request", {
+    headers: {
+      "api-key": "REDACTED"
+    }
+  });
+
+  // Test headers object with authorization header
+  const authorizationTest = {
+    authorization: "fakeValue"
+  }
+  debug("request", authorizationTest);
+  expect(spy).toHaveBeenCalledWith("OpenAI:DEBUG:request", {
+    authorization: "REDACTED"
+  });
+
+  process.env = originalEnv;
```

Comment:
`env` is not being set so I am not sure why we're resetting it here?

I think what we need to do is to capture the value of `process.env.DEBUG` before we execute the test and then assign it back here.